integrations/aws/utils.py

```diff
@@ -0,0 +1,256 @@
+    
+    
+    for credentials in _aws_credentials:
+        aws_sessions.append(credentials.createSession(custom_region if custom_region else None))
+    
+    return aws_sessions
+
+def _fix_unserializable_date_properties(obj: Any) -> Any:
+    """
+    Handles unserializable date properties in the JSON by turning them into a string
+    """
+    return json.loads(json.dumps(obj, default=str))
+
+def describe_single_resource(kind: str, identifier: str, account_id: Optional[str] = None, region: Optional[str] = None) -> dict[str, Any]:
+    """
+    Describes a single resource using the CloudControl API.
+    """
+    sessions = _get_sessions(account_id, region)
+    for session in sessions:
+        region = session.region_name
+        try:
+            if kind == ResourceKindsWithSpecialHandling.ACM:
+                acm = session.client('acm')
+                response = acm.describe_certificate(CertificateArn=identifier)
+                return response.get('Certificate', {})
+            
+            elif kind == ResourceKindsWithSpecialHandling.LOADBALANCER:
+                elbv2 = session.client('elbv2')
+                response = elbv2.describe_load_balancers(LoadBalancerArns=[identifier])
+                return response.get('LoadBalancers', [])[0]
+            
+            elif kind == ResourceKindsWithSpecialHandling.CLOUDFORMATION:
+                cloudformation = session.client('cloudformation')
+                response = cloudformation.describe_stacks(StackName=identifier)
+                return response.get('Stacks', [])[0]
+            
+            elif kind == ResourceKindsWithSpecialHandling.EC2:
+                ec2_client = session.client('ec2')
+                described_instance = ec2_client.describe_instances(InstanceIds=[identifier])
+                instance_definition = described_instance["Reservations"][0]["Instances"][0]
+                return instance_definition
+            
+            else:
+                cloudcontrol = session.client('cloudcontrol')
+                response = cloudcontrol.get_resource(TypeName=kind, Identifier=identifier)
+                resource_description = response.get('ResourceDescription', {})
+                return {
+                    'Identifier': resource_description.get('Identifier', ''),
+                    'Kind': kind,
+                    **json.loads(resource_description.get('Properties', {}))
+                }
+        except Exception as e:
+            if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                logger.info(f"Resource not found: {kind} {identifier}")
+                return {}
+            logger.error(f"Failed to describe CloudControl Instance in region: {region}; error {e}")
+            break
+
+def describe_resources(sessions: list[boto3.Session], service_name: str, describe_method: str, list_param: str, marker_param: str = "NextToken") -> ASYNC_GENERATOR_RESYNC_TYPE:
```

Comment:
lets rephrase it to be more of a `get` / `list` / `query` / `iterate` / `paginate` / `batch`, as the describe sounds to me like you provide it a resource and it returns you more description about that resource.